app/main.py

```diff
@@ -1,3 +1,125 @@
+            product_cart=customer["product_cart"],
+            location=customer["location"],
+            money=customer["money"],
+            car=Car(brand=customer["car"]["brand"],
+                    fuel_consumption=customer["car"]["fuel_consumption"], )
+        ))
+
+    shops = []
+    for shop in data["shops"]:
+        shops.append(Shop(
+            name=shop["name"],
+            products=shop["products"],
+            location=shop["location"],
+        ))
+    return {
+        "FUEL_PRICE": data["FUEL_PRICE"],
+        "customers": customers,
+        "shops": shops,
+    }
+
+
+def length_trip_calculation(x_point: list, y_point: list) -> float:
+    return math.hypot(y_point[0] - x_point[0], y_point[1] - x_point[1])
+
+
+def cost_trip_calculation(
+        length_trip: float,
+        fuel_consumption: float,
+        fuel_price: float,
+        cost_products: float,
+) -> float:
+    return round(
+        length_trip * fuel_consumption / 100
+        * fuel_price * 2 + cost_products, 2
+    )
+
+
+def shop_trip() -> None:
+    data = get_dict_from_json()
+    customers = data["customers"]
+    shops = data["shops"]
+
+    for customer in customers:
+        # amount of money
+        print(f"{customer.name} has "
+              f"{customer.money} dollars")
+
+        # path cost calculation
+        for shop in shops:
+            costs = {
+                shop.name: cost_trip_calculation(
+                    length_trip=length_trip_calculation(
+                        shop.location, customer.location
+                    ),
+                    fuel_consumption=customer.car.fuel_consumption,
+                    fuel_price=data["FUEL_PRICE"],
+                    cost_products=(shop.products["milk"]
+                                   * customer.product_cart["milk"]
+                                   + shop.products["bread"]
+                                   * customer.product_cart["bread"]
+                                   + shop.products["butter"]
+                                   * customer.product_cart["butter"]))
+                for shop in shops
+            }
+
+            print(
+                f"{customer.name}'s trip to the "
+                f"{shop.name} costs {costs[shop.name]}")
+        shop_min_cost = "".join(
+            ([k for k, v in costs.items() if v == min(costs.values())])
+        )
+        selected_shop = [
+            shop for shop in shops if shop.name == shop_min_cost
+        ][0]
+
+        if min(costs.values()) <= customer.money:
+            current_data = "04/01/2021 12:33:41"
+            milk_price = \
+                customer.product_cart["milk"] * \
+                selected_shop.products["milk"]
+            bread_price = \
+                customer.product_cart["bread"] * \
+                selected_shop.products["bread"]
+            butter_price = \
+                customer.product_cart["butter"] * \
+                selected_shop.products["butter"]
```

Comment:
No it is wrong. If we add new product you should change your code. Products can be any